### PR TITLE
[webcanvas] Implement `projxy` feature

### LINKED ISSUE #10660: TH2::SetShowProjectionXY
### Is your feature request related to a problem? Please describe.
When working with TH2 histograms, it is very common to right-click on the canvas and use SetShowProjectionX or SetShowProjectionY.

There are cases where one needs to see both of these at the same time.

### Describe the solution you'd like
A function which plots the two projections at the same time, once a rectangular window has been defined.
Ideally, one should be able to add more (gdirectory-open) TH2s to the 'buffer' via a Combo, as it is very useful to compare several 2D images by looking at their 1D projections.
one of the projections could be rotated by 90deg, if needed, to make it fit with 2D axis orientation.

### Describe alternatives you've considered
I programmed back then this tool: https://root-forum.cern.ch/t/th2-x-y-projection-interactive-viewer/26335
but it would be nice if it were adopted by ROOT, and be accesible by the context menu of the TCanvas.

### Additional context
https://stackoverflow.com/questions/40641895/plot-aligned-x-y-1d-histograms-from-projected-2d-histogram
https://www.astrobetter.com/blog/2014/02/10/visualization-fun-with-python-2d-histogram-with-1d-histograms-on-axes/

### PR DESCRIPTION
**Pull request by linev**
Allows to show both projections at a time.
Also prevents websocket overloading.
Includes jsroot with standalone feature

Resolves #10660

### CODE REVIEW

**Comment by ferdymercury** [comment]
Thanks for this!! Is the nBinsX forced to be nBinsY as in the gsi-web example you sent, or are they separately customizable?

**Comment by linev** [comment]
>  Is the nBinsX forced to be nBinsY as in the gsi-web example you sent, or are they separately customizable?

For the moment they are the same, one can provide more flexible solution in the future.

**Review by couet — APPROVED** [review]

File: js/modules/hist2d/RH2Painter.mjs [hunk 1]
```diff
@@ -81,10 +102,16 @@ class RH2Painter extends RHistPainter {
...
    fillHistContextMenu(menu) {
-      menu.add('sub:Projections', () => this.toggleProjection());
       let kind = this.is_projection || '';
-      if (kind) kind += this.projection_width;
-      let kinds = ['X1', 'X2', 'X3', 'X5', 'X10', 'Y1', 'Y2', 'Y3', 'Y5', 'Y10'];
+      if (kind) kind += this.projection_widthX;
+      if ((this.projection_widthX != this.projection_widthY) && (this.is_projection == 'XY'))
+         kind = `X${this.projection_widthX}_Y${this.projection_widthY}`;
+
+      menu.add('sub:Projections', () => menu.input('Input projection kind X1 or XY2', kind, 'string').then(val => this.toggleProjection(val)));
```
Comment (ferdymercury) [inline js/modules/hist2d/RH2Painter.mjs:110]:
```suggestion
      menu.add('sub:Projections', () => menu.input('Input projection kind X1 or XY2 or X3_Y4', kind, 'string').then(val => this.toggleProjection(val)));
```

File: js/modules/hist2d/TH2Painter.mjs [hunk 2]
```diff
@@ -160,11 +204,15 @@ class TH2Painter extends THistPainter {
...
-         const kinds = ['X1', 'X2', 'X3', 'X5', 'X10', 'Y1', 'Y2', 'Y3', 'Y5', 'Y10'];
-         if (this.is_projection) kinds.push('Off');
+         if (kind) kind += this.projection_widthX;
+         if ((this.projection_widthX != this.projection_widthY) && (this.is_projection == 'XY'))
+            kind = `X${this.projection_widthX}_Y${this.projection_widthY}`;
+
+         const kinds = ['X1', 'X2', 'X3', 'X5', 'X10', 'Y1', 'Y2', 'Y3', 'Y5', 'Y10', 'XY1', 'XY2', 'XY3', 'XY5', 'XY10'];
+         if (kind) kinds.unshift('Off');
+
+         menu.add('sub:Projections', () => menu.input('Input projection kind X1 or XY2', kind, 'string').then(val => this.toggleProjection(val)));
```
Comment (ferdymercury) [inline js/modules/hist2d/TH2Painter.mjs:215]:
```suggestion
         menu.add('sub:Projections', () => menu.input('Input projection kind X1 or XY2 or X3_Y4', kind, 'string').then(val => this.toggleProjection(val)));
```

**Review by ferdymercury — COMMENTED** [review]
Thanks so much!!

**Pull request merged**